openquake/hazardlib/correlation.py

```diff
@@ -55,6 +55,10 @@ def apply_correlation(self, sites, imt, residuals, stddev_intra=0):
         # of N random numbers (where N is equal to number of sites).
         # we need to do that multiplication once per realization
         # with the same matrix and different vectors.
+
+        if numpy.any(abs(sites.depths) > 1e-7):
+            raise Warning('Depths are not considered when dealing with spatial'
+                          ' correlation')
         try:
             corma = self.cache[imt]
         except KeyError:
```

Comment:
I am against the warning. The risk is that  it will be raised lots and lots of time flooding the log and causing a general slowdown of the engine, on top of annoying the users.